regression_tests/tests/gitbook/test_docs.py

```diff
@@ -0,0 +1,181 @@
+    else:
+        # Otherwise, we look at all files in all sections in the `docs_folder`.
+        args.sections = os.walk(args.docs_folder)
+
+    successfully_ran_all = True
+    for item in args.sections:
+        # Check to see if the section is excluded via `blacklist_sections`.
+        skip = False
+        for blacklist_section in blacklist_sections:
+            current_section = item[0].split("gitbook/")[-1]
+            if current_section == blacklist_section or current_section.startswith(blacklist_section+"/"):
+                skip = True
+                print(">> SKIPPING ", current_section)
+                break
+
+        # skip = False which means we want to run the files in the section.
+        if not skip:
+            for file in glob(os.path.join(item[0], '*.md')):
+                file_name = file.split("gitbook/")[-1]
+
+                # Check to see if the section is excluded via `blacklist_files`.
+                skip = False
+                for blacklist_file in blacklist_files:
+                    if file_name == blacklist_file:
+                        skip = True
+                        print(">> SKIPPING ", file_name)
+
+                # skip = False which means we want to run the file.
+                if not skip:
+                    snippets = get_code(file)
+                    
+                    # Remove any snippet that shows up in `blacklist_snippets`.
+                    if file_name in blacklist_snippets.keys():
+                        snippets = [snippet for snippet in snippets if snippet not in blacklist_snippets[file_name]]
+                    
+                    # If we still have code to run, run it.
+                    if len(snippets) > 0:
+                        snippet_result = run("\n".join(snippets), file.replace(r"/", "_")[:-2] + "py")
+
+                        if not snippet_result.success:
+                            # Snippet failed to run. Display relevant information for debugging. 
+                            successfully_ran_all = False
+                            print(">> FAILED   ", file_name)
+                            # Display the code
+                            print("-"*25 + "[CODE]" + "-"*25)
+                            print(snippet_result.snippet)
+                            # Display the output
+                            print("-"*25 + "[OUTPUT]" + "-"*25)
+                            print(snippet_result.output)
+                            # Display the error
+                            print("-"*25 + "[ERROR]" + "-"*25)
+                            print(snippet_result.error)
+                        else:
+                            print(">> OK       ", file_name)
+                    else:
+                        print(">> SKIPPING  (no Python snippets found)", file_name)
+
+
+    if successfully_ran_all:
+        print("Congrats! You've successfully ran all code snippets in the documentation.")
```

Comment:
If we failed, we should exit the script with some sort of exit code right? That would make it way easier to incorporate into GH actions.